test/uk/gov/hmrc/agentsubscription/model/ValidatorSpec.scala

```diff
@@ -62,7 +64,77 @@ class ValidatorSpec extends UnitSpec {
     }
   }
 
+  "overseasTelephoneNumberValidation reader" should {
+    def validateOSNumber(number: String) = overseasTelephoneNumberValidation.reads(JsString(number))
+
+    "accept input" when {
+      "there is at least 1 digit in the input" in {
+        validateOSNumber("4") shouldBe JsSuccess("4")
+      }
+      "input contains 24 or fewer digits" in {
+        validateOSNumber("123456789012345678901234") shouldBe JsSuccess("123456789012345678901234")
+      }
+
+      "contains letters" in {
+        validateOSNumber("ABCDEFGHIJKLM") shouldBe JsSuccess("ABCDEFGHIJKLM")
+        validateOSNumber("NOPQRSTUVWXYZ") shouldBe JsSuccess("NOPQRSTUVWXYZ")
```

Comment:
these could be lowercase since youre double checking or mix of lower and upper case

Reply:
It turns out that only uppercase letter are allowed so I added a test for the lowercase letters being rejected.